dotnet/src/Agents/OpenAI/Internal/AssistantThreadActions.cs

```diff
@@ -354,57 +374,58 @@ private static ChatMessageContent GenerateMessageContent(string? assistantName,
                 AuthorName = assistantName,
             };
 
-        foreach (MessageContent itemContent in message.ContentItems)
+        foreach (MessageContent itemContent in message.Content)
         {
             // Process text content
-            if (itemContent is MessageTextContent contentMessage)
+            if (!string.IsNullOrEmpty(itemContent.Text))
             {
-                content.Items.Add(new TextContent(contentMessage.Text.Trim()));
+                content.Items.Add(new TextContent(itemContent.Text.Trim()));
 
-                foreach (MessageTextAnnotation annotation in contentMessage.Annotations)
+                foreach (TextAnnotation annotation in itemContent.TextAnnotations)
                 {
                     content.Items.Add(GenerateAnnotationContent(annotation));
                 }
             }
             // Process image content
-            else if (itemContent is MessageImageFileContent contentImage)
+            else if (itemContent.ImageFileId != null)
             {
-                content.Items.Add(new FileReferenceContent(contentImage.FileId));
+                content.Items.Add(new FileReferenceContent(itemContent.ImageFileId));
```

Comment:
Are we unable to use `ImageContent` here? Wondering why it needs to be `FileReferenceContent`?

Reply:
I see we're dealing with an image file file_id, so I guess that's why...

Reply:
There is no `ImageContent.ImageFileId`.  Since `ImageContent` inherits from `BinaryContent` and has been "graduated", I didn't want to churn this contract.

In the response case, if an image is provided by the assistant (or any other file), we represent that as a `FileReferenceContent`.